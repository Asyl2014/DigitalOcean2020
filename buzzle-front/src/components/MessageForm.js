import React from 'react';

class MessageForm extends React.Component {
    state = {
        'content': ''
    };

    handleChange = event => {
        this.setState({ 'content': event.target.value });
    }

    handleSubmit = event => {
        event.preventDefault();

        const newContent = this.state.content;
        this.props.handleMessageCreate(newContent, () => {
            this.setState({ 'content': '' });
        });
    }

    render() {
        return (
            <div className="row justify-content-center">
                <div className="col-md-6 px-5 mt-4">
                    <form action="/message/create" method="POST" onSubmit={this.handleSubmit} >
                        <label htmlFor="message-content" className="text-primary">
                            <strong>Whatâ€™s on your mind?</strong>
                        </label>
                        <div className="input-group mb-3">
                            <input
                                id="message-content"
                                className="form-control text-white border-primary bg-primary"
                                type="text"
                                name="content"
                                autoComplete="off"
                                onChange={this.handleChange}
                                value={this.state.content} />
                            <div className="input-group-append">
                                <input className="btn btn-primary" type="submit" value="Send" />
                            </div>
                        </div>
                    </form>
                </div>
            </div>
        );
    }
}

export default MessageForm;
