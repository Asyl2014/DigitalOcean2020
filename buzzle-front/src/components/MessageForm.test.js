import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import MessageForm from './MessageForm.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageForm />, div);
});

it('renders two messages', () => {
    const wrapper = mount(<MessageForm />);
    const input = wrapper.find('#message-content');
    input.simulate('change', { target: { value: 'hello' }});
    expect(wrapper.state().content).toBe('hello');
});

it('renders correctly and equal to snapshot', () => {
    const tree = renderer.create(<MessageForm />)
                         .toJSON();

    expect(tree).toMatchSnapshot();
});