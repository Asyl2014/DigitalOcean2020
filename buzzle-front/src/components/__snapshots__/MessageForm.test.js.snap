// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and equal to snapshot 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-md-6 px-5 mt-4"
  >
    <form
      action="/message/create"
      method="POST"
      onSubmit={[Function]}
    >
      <label
        className="text-primary"
        htmlFor="message-content"
      >
        <strong>
          Whatâ€™s on your mind?
        </strong>
      </label>
      <div
        className="input-group mb-3"
      >
        <input
          autoComplete="off"
          className="form-control text-white border-primary bg-primary"
          id="message-content"
          name="content"
          onChange={[Function]}
          type="text"
          value=""
        />
        <div
          className="input-group-append"
        >
          <input
            className="btn btn-primary"
            type="submit"
            value="Send"
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;
